File: src/test/java/kitchenpos/domain/FakeMenuGroupRepository.java
package kitchenpos.domain;

import kitchenpos.menus.domain.MenuGroup;
import kitchenpos.menus.domain.MenuGroupRepository;

import java.util.*;

public class FakeMenuGroupRepository implements MenuGroupRepository {

    private final HashMap<UUID, MenuGroup> inMemory = new HashMap<>();

    @Override
    public MenuGroup save(MenuGroup menuGroup) {
        inMemory.put(menuGroup.getId(), menuGroup);
        return menuGroup;
    }

    @Override
    public Optional<MenuGroup> findById(UUID id) {
        return Optional.ofNullable(inMemory.get(id));
    }

    @Override
    public List<MenuGroup> findAll() {
        return new ArrayList<>(inMemory.values());
    }
}

File: src/test/kotlin/kitchenpos/menu/tobe/infra/FakeMenuGroupRepository.kt
package kitchenpos.menu.tobe.infra

import java.util.*
import kitchenpos.menu.tobe.domain.MenuGroup
import kitchenpos.menu.tobe.domain.MenuGroupRepository

class FakeMenuGroupRepository : MenuGroupRepository {
    private val menuGroups: MutableMap<UUID, MenuGroup> = mutableMapOf()

    override fun save(menuGroup: MenuGroup): MenuGroup {
        menuGroup.id = UUID.randomUUID()
        menuGroups[menuGroup.id!!] = menuGroup
        return menuGroup
    }

    override fun findById(id: UUID): Optional<MenuGroup> {
        return Optional.ofNullable(menuGroups[id])
    }

    override fun findAll(): List<MenuGroup> {
        return menuGroups.values.toList()
    }
}
